### ISSUE TITLE
There have been so many reports from users that want to eval code from random context into whatever their most recent REPL is that we should probably add an option to allow this.

### ISSUE THREAD

**Issue by blak3mill3r** [clojure-emacs/cider#2464]
There's at least 1 person complaining about the changes we've made every day. :-) 

For that particular issue I agree with your second suggestion.

_Originally posted by @bbatsov in https://github.com/clojure-emacs/cider/issues/2446#issuecomment-420600951_

**Comment by blak3mill3r** [comment]
Opening this issue for those interested in an option to mimic the behavior of previous versions of CIDER, in which code is `eval`d in the "default repl".

I find the new behavior frustrating because I run many separate instances of emacs, each one connected to zero or one nREPL connections.

That means I never had any confusion about where the code was being evaluated, and could easily eval code in a checkout or even just a fork of some dependency.

Such an option would be useful for those of us who launch separate Emacsen for separate projects.

**Comment by blak3mill3r** [comment]
I found a workaround that does what I want. YMMV.

In my `cider-mode-hook` I run this:

```clojure
    (sesman-link-with-buffer (current-buffer) (car (hash-table-values sesman-sessions-hashmap)))
```

and it seems to behave like the old CIDER.

**Comment by bbatsov** [comment]
Yeah, I completely agree we should add some extra configuration for this. Seems we just failed to anticipate all the ways in which people were using CIDER. In the classic style of software engineering - we managed to solve a problem for some of the users, by creating a problem for another part of the users. 😆 

Anyways, I believe that some behaviour of the type "if there's no session for this context just use the most recent session" should be very easy to implement. I'm taking a break from CIDER to focus on some other things, but I'm hoping that @vspinu or someone else will be able to take a look at this soon. 
Afterwards I'd be happy to cut a 0.18.1 release.

**Comment by blak3mill3r** [comment]
Okay, that description helps me grok what's relevant, perhaps I can add such an option. I'm happy with my `cider-connect-mode-hook` hack for right now and I can't work on this today but I'll come back to it.

Thank you for making CIDER so fun and useful @bbatsov ... :beers:

**Comment by vspinu** [comment]
I am traveling till 9th. Will have a look then. The automatic linking of
dependency projects should go a long way in solving this. I will also add
an option to sesman to pick first available repl if no link is in place.

On 30 Sep 2018 20:51, "Blake Miller" <notifications@github.com> wrote:

Okay, that description helps me grok what's relevant, perhaps I can add
such an option. I'm happy with my cider-connect-mode-hook hack for right
now and I can't work on this today but I'll come back to it.

Thank you for making CIDER so fun and useful @bbatsov
<https://github.com/bbatsov> ... 🍻

—
You are receiving this because you were mentioned.

Reply to this email directly, view it on GitHub
<https://github.com/clojure-emacs/cider/issues/2464#issuecomment-425742576>,
or mute the thread

**Comment by bbatsov** [comment]
Sounds like a plan to me!
